xla/backends/cpu/benchmarks/e2e/gemma2/keras/benchmark.py

```diff
@@ -51,21 +51,34 @@ def compute_stats(array):
   return (mean, diff)
 
 
-def run(gemma_lm, max_len):
+def run(gemma_lm, tokenizer, max_len):
   """Benchmarks inferences with at most `max_len` output tokens.
 
   Args:
     gemma_lm: The Gemma2 Keras model.
+    tokenizer: The tokenizer for the Gemma2 model.
     max_len: The maximum number of output tokens per one inference.
 
   Returns:
     mean ± %diff and the actual number of output tokens generated per inference.
   """
+  in_tokens = tokenizer(_QUERY)
+  in_tokens_len = len(in_tokens)
+  total_output_len = in_tokens_len + max_len
+  if max_len < 1:
+    print(f"Error: max_len {max_len} should be >= 1")
+    exit()
+
   # Warm up.
   start = time.time()
-  output = gemma_lm.generate(_QUERY, max_length=max_len + 1)
-  num_actual_output_tokens = len(output.split(" "))
+  output = gemma_lm.generate(_QUERY, max_length=total_output_len + 1)
   warmup_time = (time.time() - start) * 1000
+  num_actual_output_tokens = len(tokenizer(output))
+  gen_tokens = 0
+  if num_actual_output_tokens > in_tokens_len:
+    gen_tokens = num_actual_output_tokens - in_tokens_len
```

Comment:
Nit: We can make this more concise
```suggestion
  gen_tokens = max(num_actual_output_tokens - in_tokens_len, 0)
```